libs/wire-api/src/Wire/API/User/RichInfo.hs

```diff
@@ -290,30 +295,24 @@ modelRichField = Doc.defineModel "RichField" $ do
   Doc.property "value" Doc.string' $
     Doc.description "Field value"
 
-instance ToJSON RichField where
-  -- NB: "name" would be a better name for 'richFieldType', but "type" is used because we
-  -- also have "type" in SCIM; and the reason we use "type" for SCIM is that @{"type": ...,
-  -- "value": ...}@ is how all other SCIM payloads are formatted, so it's quite possible
-  -- that some provisioning agent would support "type" but not "name".
-  toJSON u =
-    object
-      [ "type" .= CI.original (richFieldType u),
-        "value" .= richFieldValue u
-      ]
-
-instance FromJSON RichField where
-  parseJSON = withObject "RichField" $ \o -> do
-    RichField
-      <$> (CI.mk <$> o .: "type")
-      <*> o .: "value"
-
 instance Arbitrary RichField where
   arbitrary =
     RichField
       <$> (CI.mk . cs . QC.getPrintableString <$> arbitrary)
       <*> (cs . QC.getPrintableString <$> arbitrary)
   shrink (RichField k v) = RichField <$> QC.shrink k <*> QC.shrink v
 
+instance S.ToSchema RichField where
+  -- NB: "name" would be a better name for 'richFieldType', but "type" is used because we
+  -- also have "type" in SCIM; and the reason we use "type" for SCIM is that @{"type": ...,
+  -- "value": ...}@ is how all other SCIM payloads are formatted, so it's quite possible
+  -- that some provisioning agent would support "type" but not "name".
+  schema =
+    S.object "RichField" $
+      RichField
+        <$> richFieldType S..= S.field "type" (S.dimap CI.original CI.mk S.schema)
```

Comment:
Same here about `dimap`: `CI.original .= CI.mk <$> schema` is how I normally write these things.